### ISSUE TITLE
[helm] Provide a global and consistent floatingUserId

### ISSUE THREAD

**Issue by lgadban** [solo-io/gloo#5034]
**Is your feature request related to a problem? Please describe.**
to install on OCP, you have to enable floatingUserId on tons of components. it is very error-prone and provides a poor UX

**Describe the solution you'd like**
A global way to turn on floatingUserId (and also have it include things like fsGroup)

This relates to part 2A of https://github.com/solo-io/solo-projects/issues/6323

Part of the scope of this ticket includes considering how this applies to the Kubes GW helm charts, which currently do not support `floatingUserId`, getting consensus on an approach, and implementing the decided-upon approach, if any.

**Additional context**
This is what a values file looks like to install on OCP:
```
global:
  extensions:
    extAuth:
      deployment:
        floatingUserId: true
        fsGroup: 1001850000
    rateLimit:
      deployment:
        floatingUserId: true
gloo:
  gloo:
    deployment:
      floatingUserId: true
  discovery:
    deployment:
      floatingUserId: true
      fsGroup: 1001850000
  gateway:
    deployment:
      floatingUserId: true
    certGenJob:
      floatingUserId: true  
  gatewayProxies:
    gatewayProxy:
      podTemplate:
        floatingUserId: true
        fsGroup: 1001850000
redis:
  deployment:
    floatingUserId: true
    fsGroup: 1001850000
observability:
  deployment:
    floatingUserId: true
    fsGroup: 1001850000
```

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1206768562311555/1207720937031997) by [Unito](https://www.unito.io)

**Comment by chrisgaun** [comment]
Duplicate of https://github.com/solo-io/gloo/issues/4196

**Comment by lgadban** [comment]
@chrisgaun this is not a dupe of the linked issue (https://github.com/solo-io/gloo/issues/4196)

this issue is to improve the helm values UX by providing a single global value that would set the `floatingUserId` for all deployments.

The use case is that this value is typically used for OpenShift; in which case, you would need to set it for every single deployment. This can be made extremely more simple by using a single global value.

**Comment by nrjpoddar** [comment]
@sam-heilbron to look at this issue for scope/estimation.

**Comment by sheidkamp** [comment]
Proposing adding the following helm values:

```
|global.securityDefaults.floatingUserId|bool||If true, use 'true' as default value for all instances of floatingUserId. Can not be overriden locally.|
|global.securityDefaults.runAsUser|float64||Sets the default 'runAsUser' value for all instances of runAsUser. May be overriden locally|
|global.securityDefaults.fsGroup|float64||Sets the default 'fsGroup' value for all instances of fsGroup. May be overriden locally|
```

which can be used in the templates like:
```
        {{ $glooContainerSecurityValues := merge (or .Values.gloo.deployment.glooContainerSecurityContext dict) (or .Values.global.securityDefaults dict)}}
        {{- if not $glooContainerSecurityValues.floatingUserId -}}
          {{- $_ := set $securityDefaults "runAsUser" .Values.gloo.deployment.runAsUser }}
        {{- end -}}
        {{- include "gloo.containerSecurityContext" (dict "values" $glooContainerSecurityValues "defaults" $securityDefaults "podSecurityStandards" .Values.global.podSecurityStandards "indent" 8) }}
```

replacing

```
        {{- if not .Values.gloo.deployment.floatingUserId -}}
          {{- $_ := set $securityDefaults "runAsUser" .Values.gloo.deployment.runAsUser }}
        {{- end -}}
        {{- include "gloo.containerSecurityContext" (dict "values" .Values.gloo.deployment.glooContainerSecurityContext "defaults" $securityDefaults "podSecurityStandards" .Values.global.podSecurityStandards "indent" 8) }}
```

**Comment by sheidkamp** [comment]
Clarification of scope:
- Includes GlooPortal templates
- Includes parts 2a/2b/3 of the [OpenShift Support Epic](https://github.com/solo-io/solo-projects/issues/6323) - we will introduce `floatingUserId` to GatewayParams
- Includes places where `floatingUserId` is applied, but the securityContext template is not currently used (in solo-projects the extauth pod and the apiserver-deployment in gloo-fed.
- Includes extra bits of logic in solo-projects' redis template to make setting `redis.deployment.enablePodSecurityContext=false` unnecessary in OpenShift

**Comment by sheidkamp** [comment]
Implemented in main and 1.17.x and ready for 1.17.1 release
